core/templates/dev/head/domain/topic/TopicUpdateServiceSpec.js

```diff
@@ -410,23 +441,105 @@ describe('Topic update service', function() {
   });
 
   it('should move a skill id to a subtopic', function() {
-    expect(_sampleTopic.getUncategorizedSkillIds()).toEqual(['skill_1']);
-    expect(_sampleTopic.getSubtopics()[0].getSkillIds()).toEqual(['skill_2']);
-    TopicUpdateService.moveSkillIdToSubtopic(_sampleTopic, null, 1, 'skill_1');
-    expect(_sampleTopic.getUncategorizedSkillIds()).toEqual([]);
-    expect(_sampleTopic.getSubtopics()[0].getSkillIds()).toEqual([
-      'skill_2', 'skill_1'
+    expect(_sampleTopic.getUncategorizedSkillSummaries()).toEqual([
+      _firstSkillSummary
+    ]);
+    expect(_sampleTopic.getSubtopics()[0].getSkillSummaries()).toEqual([
+      _secondSkillSummary
+    ]);
+    TopicUpdateService.moveSkillToSubtopic(
+      _sampleTopic, null, 1, _firstSkillSummary);
+    expect(_sampleTopic.getUncategorizedSkillSummaries()).toEqual([]);
+    expect(_sampleTopic.getSubtopics()[0].getSkillSummaries()).toEqual([
+      _secondSkillSummary, _firstSkillSummary
     ]);
 
     UndoRedoService.undoChange(_sampleTopic);
-    expect(_sampleTopic.getUncategorizedSkillIds()).toEqual(['skill_1']);
-    expect(_sampleTopic.getSubtopics()[0].getSkillIds()).toEqual(['skill_2']);
+    expect(_sampleTopic.getUncategorizedSkillSummaries()).toEqual([
+      _firstSkillSummary
+    ]);
+    expect(_sampleTopic.getSubtopics()[0].getSkillSummaries()).toEqual([
+      _secondSkillSummary
+    ]);
+  });
+
+  it('should correctly create changelists when moving a skill to a new ' +
```

Comment:
Is this the correct description? Seems to me like you're testing various "delete subtopic" cases.

Reply:
Yeah, I am testing the delete subtopic cases following a move operation through it. Will update the name to this, then.

Reply:
Done.